portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/security/security/usecase/SecurityUsecase.testcase

```diff
@@ -41,11 +43,9 @@ definition {
 		}
 
 		task ("Assert message present in HTML source") {
-			Pause(locator1 = "3000");
-
-			User.viewWelcomeMEssageHTML(
-				userfn = "${userfn}",
-				userln = "${userln}");
+			while (IsElementPresent(locator1 = "JSONWS#SECURITY_RESULT", value1 = "Welcome ${userfn} ${userln}!")) {
```

Comment:
Do we have a timeout for while loops?

Reply:
It has a default value how much it will do the while loop but I need to look it up. thouh we can use **maxIterations** to set how many times it can do it manually

Reply:
True: https://github.com/liferay/liferay-portal/blob/master/modules/test/poshi/poshi-runner/src/main/java/com/liferay/poshi/runner/PoshiRunnerExecutor.java#L1028